src/com/facebook/buck/apple/project_generator/ProjectGenerator.java

```diff
@@ -1608,7 +1627,27 @@ public static String getProductName(TargetNode<?> buildTargetNode, BuildTarget b
           CxxPlatformXcodeConfigGenerator.getDefaultXcodeBuildConfigurationsFromCxxPlatform(
               defaultCxxPlatform,
               appendedConfig);
-      configs = Optional.of(ImmutableSortedMap.copyOf(defaultConfig));
+
+      if (isFrameworkTarget(targetNode)) {
+        ImmutableMap.Builder<String, ImmutableMap<String, String>> configBuilder =
+            ImmutableMap.builder();
+        for (Map.Entry<String, ImmutableMap<String, String>> entry : defaultConfig.entrySet()) {
+          ImmutableMap.Builder<String, String> frameworkConfigs = ImmutableMap.builder();
+          frameworkConfigs.putAll(entry.getValue());
+
+          frameworkConfigs.put("DYLIB_INSTALL_NAME_BASE", "@rpath");
+          frameworkConfigs.put("SKIP_INSTALL", "YES");
+          frameworkConfigs.put(
+              "LD_RUNPATH_SEARCH_PATHS",
+              "$(inherited) @executable_path/Frameworks @loader_path/Frameworks");
+          frameworkConfigs.put("DEFINES_MODULE", "YES");
+
+          configBuilder.put(entry.getKey(), frameworkConfigs.build());
+        }
+        configs = Optional.of(ImmutableSortedMap.copyOf(configBuilder.build()));
```

Comment:
No, that was a mistake, I've updated this to be a little bit clearer and also to set `LD_RUNPATH_SEARCH_PATHS` by default.